app/controllers/internal_api/v1/pillars_controller.rb

```diff
@@ -0,0 +1,43 @@
+class InternalApi::V1::PillarsController < InternalApiController
+
+  def show
+    ok root: "pillar", content: {
+             dashboard: Pillar.value(pillar: :dashboard),
+             api: {
+               server: {
+                 external_fqdn: Pillar.value(pillar: :apiserver)
+               },
+               cluster_ip: Pillar.value(pillar: :api_cluster_ip)
+             },
+             cluster_cidr: Pillar.value(pillar: :cluster_cidr),
+             cluster_cidr_min: Pillar.value(pillar: :cluster_cidr_min),
+             cluster_cidr_max: Pillar.value(pillar: :cluster_cidr_max),
+             cluster_cidr_len: Pillar.value(pillar: :cluster_cidr_len),
+             services_cidr: Pillar.value(pillar: :services_cidr),
+             dns: {
+               cluster_ip: Pillar.value(pillar: :dns_cluster_ip)
+             },
+             proxy: {
+               systemwide: Pillar.value(pillar: :proxy_systemwide),
+               http: Pillar.value(pillar: :http_proxy),
+               https: Pillar.value(pillar: :https_proxy),
+               no_proxy: Pillar.value(pillar: :no_proxy)
+             },
+             ldap: {
+               host: Pillar.value(pillar: :ldap_host),
+               port: Pillar.value(pillar: :ldap_port),
+               bind_dn: Pillar.value(pillar: :ldap_bind_dn),
+               bind_pw: Pillar.value(pillar: :ldap_bind_pw),
+               domain: Pillar.value(pillar: :ldap_domain),
+               group_dn: Pillar.value(pillar: :ldap_group_dn),
+               people_dn: Pillar.value(pillar: :ldap_people_dn),
+               base_dn: Pillar.value(pillar: :ldap_base_dn),
+               admin_group_dn: Pillar.value(pillar: :ldap_admin_group_dn),
+               admin_group_name: Pillar.value(pillar: :ldap_admin_group_name),
+               tls_method: Pillar.value(pillar: :ldap_tls_method),
+               mail_attribute: Pillar.value(pillar: :ldap_mail_attribute)
```

Comment:
although it's probably quite efficient already, we're doing many db queries here, do you think it would enhance performance a bit if we cache all pillars in a variable and generate the response from that variable?

Reply:
I was thinking in improving this, yes. The pillar would invalidate the cache when anything is saved. Updating.